app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-smart-answer-results-tracker.js

```diff
@@ -0,0 +1,55 @@
+window.GOVUK = window.GOVUK || {}
+window.GOVUK.Modules = window.GOVUK.Modules || {}
+window.GOVUK.analyticsGa4 = window.GOVUK.analyticsGa4 || {};
+
+(function (Modules) {
+  'use strict'
+
+  function Ga4SmartAnswerResultsTracker (module) {
+    this.module = module
+  }
+
+  Ga4SmartAnswerResultsTracker.prototype.init = function () {
+    var consentCookie = window.GOVUK.getConsentCookie()
+
+    if (consentCookie && consentCookie.settings) {
+      this.startModule()
+    } else {
+      this.startModule = this.startModule.bind(this)
+      window.addEventListener('cookie-consent', this.startModule)
+    }
+  }
+
+  // triggered by cookie-consent event, which happens when users consent to cookies
+  Ga4SmartAnswerResultsTracker.prototype.startModule = function () {
+    if (window.dataLayer) {
+      this.trackResults()
+      this.module.addEventListener('click', this.handleClick.bind(this))
+    }
+  }
+
+  Ga4SmartAnswerResultsTracker.prototype.trackResults = function () {
+    var ecommerceSchema = GOVUK.analyticsGa4.core.ecommerceHelperFunctions.populateEcommerceSchema({
+      element: this.module,
+      resultsId: 'result-count'
```

Comment:
Would this benefit from allowing the `resultsId` to be passed as a data attribute? e.g.

```
function Ga4SmartAnswerResultsTracker (module) {
  this.module = module
  this.resultsId = this.module.getAttribute('data-ga4-results-id') || 'result-count'
}
```

Reply:
Getting the results from an attribute could get tricky because in the finder applications, you can update the search results via an AJAX request (therefore changing the number of results) and if we put the result count in a data attribute, it won't be updated accordingly. We'd need to write additional JS to handle that. Could we keep this as it is for now and then I can look at addressing this as part of the wider task of looking at naming conventions?